### PR TITLE
Shim DMA Extension For ObjectFIFOs

### PR DESCRIPTION
**Pull request by AndraBisca**
Object FIFOs can now be used to establish dataflow between AIE tiles and shim tiles.

Because shim tiles use AIE.external_buffers, a new operation has been added to the dialect: AIE.objectFifo.registerExternalBuffers. This allows to register which external buffers should be used in the shim DMA corresponding to a shim tile source/destination of an objectFifo.

With the addition of this feature on top of the recent broadcast extension, broadcast can now be done from one shim tile source to many AIE tile destinations.

### CODE REVIEW

File: include/aie/AIE.td [hunk 1]
```diff
@@ -1603,6 +1603,40 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectFifo.createObjectFifo", []> {
...
+def AIE_ObjectFifoRegisterExternalBuffersOp: AIE_Op<"objectFifo.registerExternalBuffers", []> {
+  let summary = "Registers external buffers to given object fifo to use in the shim tile";
+  let description = [{
+    The "aie.objectFifo.registerExternalBuffers" operation 
+
+    Example:
+    ```
+      %of_t70_t73 = AIE.objectFifo.createObjectFifo(%t70, %t73, 2) : !AIE.objectFifo<memref<64xi16>>
+      %buffer_in_0  = AIE.external_buffer 0x020100004000 : memref<512 x i16>
+      %buffer_in_1  = AIE.external_buffer 0x020100006000 : memref<512 x i16>
```
Comment (jgmelber) [inline include/aie/AIE.td:1615]: Did you mean to include the physical address here? #183

Comment (AndraBisca) [reply include/aie/AIE.td:1615]: No, that's a leftover from before it was removed. Thank you for pointing it out!

**Comment by AndraBisca** [comment]
Converted to draft while rewriting some tests and changing directory structure in the MM_2x2 design.

**Review by denolf — APPROVED** [review]
Looking good.

**Pull request merged**